src/coreComponents/common/MemoryInfos.cpp

```diff
@@ -27,8 +27,13 @@ MemoryInfos::MemoryInfos( umpire::MemoryResourceTraits::resource_type resourceTy
     case umpire::MemoryResourceTraits::resource_type::pinned:
       #if defined( _SC_PHYS_PAGES ) && defined( _SC_PAGESIZE )
       m_totalMemory = sysconf( _SC_PHYS_PAGES ) * sysconf( _SC_PAGESIZE );
+      #if defined(_SC_AVPHYS_PAGES)
       m_availableMemory = sysconf( _SC_AVPHYS_PAGES ) * sysconf( _SC_PAGESIZE );
       #else
```

Comment:
Would something like this be a way to find the available memory on MaxOS? Unfortunately, I we don't have any MacOS to test that here.

```suggestion
      #elif /* macos compilation */
        {
          uint64_t mem;
          size_t len = sizeof( mem );
          sysctlbyname( "hw.memsize", &mem, &len, nullptr, 0 );
          m_availableMemory= mem / sysconf(_SC_PAGE_SIZE);
        }
```

Should we regroup the system-dependant component like this in a namespace ?